pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -0,0 +1,213 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta1"
+	flowcontrolapisv1beta1 "k8s.io/kubernetes/pkg/apis/flowcontrol/v1beta1"
+)
+
+var (
+	errObjectNotFlowSchema = errors.New("object is not a FlowSchema type")
+)
+
+// FlowSchemaEnsurer ensures the specified bootstrap configuration objects
+type FlowSchemaEnsurer interface {
+	Ensure([]*flowcontrolv1beta1.FlowSchema) error
+}
+
+// FlowSchemaRemover removes the specified bootstrap configuration objects
+type FlowSchemaRemover interface {
+	Remove([]*flowcontrolv1beta1.FlowSchema) error
+}
+
+// NewSuggestedFlowSchemaEnsurer returns a FlowSchemaEnsurer instance that
+// can be used to ensure a set of suggested FlowSchema configuration objects.
+// shouldCreate indicates whether a missing 'suggested' FlowSchema object should be recreated.
+func NewSuggestedFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface, shouldCreate bool) FlowSchemaEnsurer {
+	wrapper := &flowSchemaWrapper{
+		client: client,
+	}
+	return &fsEnsurer{
+		strategy: newSuggestedEnsureStrategy(wrapper, shouldCreate),
+		wrapper:  wrapper,
+	}
+}
+
+// NewMandatoryFlowSchemaEnsurer returns a FlowSchemaEnsurer instance that
+// can be used to ensure a set of mandatory FlowSchema configuration objects.
+func NewMandatoryFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface) FlowSchemaEnsurer {
+	wrapper := &flowSchemaWrapper{
+		client: client,
+	}
+	return &fsEnsurer{
+		strategy: newMandatoryEnsureStrategy(wrapper),
+		wrapper:  wrapper,
+	}
+}
+
+// NewFlowSchemaRemover returns a FlowSchemaRemover instance that
+// can be used to remove a set of FlowSchema configuration objects.
+func NewFlowSchemaRemover(client flowcontrolclient.FlowSchemaInterface) FlowSchemaRemover {
+	return &fsEnsurer{
+		wrapper: &flowSchemaWrapper{
+			client: client,
+		},
+	}
+}
+
+// GetFlowSchemaRemoveCandidate returns a list of FlowSchema objects
+// that are candidates for deletion from the cluster.
+// bootstrap: a set of hard coded FlowSchema configuration objects
+// kube-apiserver maintains in-memory.
+func GetFlowSchemaRemoveCandidate(client flowcontrolclient.FlowSchemaInterface, bootstrap []*flowcontrolv1beta1.FlowSchema) ([]*flowcontrolv1beta1.FlowSchema, error) {
+	// TODO: use a lister here to avoid periodic LIST calls
+	fsList, err := client.List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list FlowSchema - %w", err)
+	}
+
+	bootstrapObjects := map[string]metav1.Object{}
+	for i := range bootstrap {
+		object := bootstrap[i]
+		bootstrapObjects[object.GetName()] = bootstrap[i]
+	}
+
+	currentObjects := make([]metav1.Object, len(fsList.Items))
+	for i := range fsList.Items {
+		currentObjects[i] = &fsList.Items[i]
+	}
+
+	list := getRemoveCandidate(bootstrapObjects, currentObjects)
+	candidates := make([]*flowcontrolv1beta1.FlowSchema, len(list))
```

Comment:
And here you won't need this conversion, because getRemoveCandidate can already remove the []string (which you just remove here).